### ISSUE TITLE
torch.nn.Functional.softplus() crashes when running on 'mps'

### ISSUE THREAD

**Issue by jbpacker** [pytorch/pytorch#99428]
### 🐛 Describe the bug

Computer hardware overview:

```
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro18,3
  Model Number:	MKGP3LL/A
  Chip:	Apple M1 Pro
  Total Number of Cores:	8 (6 performance and 2 efficiency)
  Memory:	16 GB
  System Firmware Version:	8419.41.10
  OS Loader Version:	8419.41.10
```

When running the following code:

```python
import torch
import torch.nn.functional as F

def main():
    x = torch.ones(10, 1).to(torch.device('mps'))
    x = F.softplus(x)
    

if __name__ == "__main__":
    main()
```

I get the error:

```
% python3.10 softplus_error.py 
loc("mps_multiply"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/810eba08-405a-11ed-86e9-6af958a02716/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":228:0)): error: input types 'tensor<10x1xf32>' and 'tensor<1xsi64>' are not broadcast compatible
LLVM ERROR: Failed to infer result type(s).
zsh: abort      python3.10 softplus_error.py
```

### Versions

The above script wasn't working for me - 404: Not found

However, here's my version of pytorch:

```
% python3.10
Python 3.10.10 (main, Feb 16 2023, 02:49:39) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.__version__
'1.13.1'
```

UPDATE:  I found the script at a different URL, here's the output:
```
python3.10 collect_env.py 
Collecting environment information...
PyTorch version: 1.13.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.0.1 (arm64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: version 3.25.3
Libc version: N/A

Python version: 3.10.10 (main, Feb 16 2023, 02:49:39) [Clang 14.0.0 (clang-1400.0.29.202)] (64-bit runtime)
Python platform: macOS-13.0.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] pytorch-lightning==1.8.4.post0
[pip3] torch==1.13.1
[pip3] torch-tb-profiler==0.4.0
[pip3] torchmetrics==0.11.0
[pip3] torchvision==0.14.0
[conda] Could not collect
```

FYI - I don't have conda

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by qqaatw** [comment]
Hi, it should be fixed by #88555. Please try the v2.0.0 release or the latest nightly.

**Comment by jbpacker** [comment]
@qqaatw @malfet Thank you for the quick update! This resolved the issue for me.
